http-api/src/main/java/io/avaje/http/api/MappedParam.java

```diff
@@ -0,0 +1,30 @@
+package io.avaje.http.api;
+
+import static java.lang.annotation.ElementType.MODULE;
+import static java.lang.annotation.ElementType.PACKAGE;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.SOURCE;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/** Marks a type to be mapped. */
+@Retention(RetentionPolicy.CLASS)
+@Target({ElementType.TYPE})
+public @interface MappedParam {
+
+  /** Factory method name used to construct the type. Empty means use a constructor */
+  String factoryMethod() default "";
+
+  @Retention(SOURCE)
+  @Target({TYPE, PACKAGE, MODULE})
+  @interface Import {
```

Comment:
I think we additionally could add a repeatable for `@MappedParam.Import` ... as we'd likely specify all those together in one place.